packages/core/src/App/Containers/QuestionnaireModal/questionnarie-modal.tsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Analytics } from '@deriv/analytics';
+import { Modal, Text } from '@deriv/components';
+import './questionnarie-modal.scss';
+
+type TAnswers = {
+    code: string;
+    text: string;
+    header?: string;
+};
+type TABQuestionnarie = {
+    id: string;
+    question: string;
+    answers: TAnswers[];
+};
+type TQuestionnarieModal = {
+    ab_questionnarie: TABQuestionnarie;
+    onSignupComplete: () => void;
+};
+const ModalContent = ({ ab_questionnarie, onSignupComplete }: TQuestionnarieModal) => {
+    const a_variant = !!ab_questionnarie.answers?.[0]?.header;
+
+    const onClickAnswer = (answer_code: string) => {
+        Analytics.trackEvent('ce_questionnaire_form', { action: 'choose_answer', answer_code });
+        onSignupComplete();
+    };
+    return (
+        <React.Fragment>
+            <Text as='h2' size='xs' weight='bold' align='center'>
+                {ab_questionnarie.question}
+            </Text>
+            <ul
+                className={classNames({
+                    'questionnarie-modal__answers': a_variant,
+                    'questionnarie-modal__options': !a_variant,
+                })}
+            >
+                {ab_questionnarie.answers.map(({ code, text, header }) => {
+                    return (
+                        <li
+                            key={`${code}_questionnarie`}
+                            className={classNames({
+                                'questionnarie-modal__answers_content': a_variant,
+                                'questionnarie-modal__options_card': !a_variant,
+                            })}
+                            onClick={() => onClickAnswer(code)}
+                        >
+                            {header && (
+                                <Text as='p' size='xs' weight='bold'>
+                                    {header}
+                                </Text>
+                            )}
+                            <Text as='p' size='xs'>
+                                {text}
+                            </Text>
+                        </li>
+                    );
+                })}
+            </ul>
+        </React.Fragment>
+    );
+};
+
+const QuestionnarieModal = ({ ab_questionnarie, onSignupComplete }: TQuestionnarieModal) => {
```

Comment:
please place this component in a separate file, and add tests for it. test coverage is a requirement when adding new components in deriv-app. :)

Reply:
done, good point